lib/src/domain/entity/movie_entity.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:flutter/foundation.dart';
+
+import '../../data/model/movie_model.dart';
+
+class MovieEntity {
+  static const String imageBaseUrl = 'https://image.tmdb.org/t/p/w500';
+  final int id;
+  final String title;
+  final String originalTitle;
+  final String overview;
+  final DateTime releaseDate;
+  final double voteAverage;
+  final List<int> genres;
+  final String poster;
+  final String backdrop;
+  final String originalLanguage;
+  final double popularity;
+  final bool hasVideo;
+  final int voteCount;
+  final bool isAdult;
+
+  MovieEntity({
+    required this.id,
+    required this.title,
+    required this.originalTitle,
+    required this.overview,
+    required this.releaseDate,
+    required this.voteAverage,
+    required this.genres,
+    required this.poster,
+    required this.backdrop,
+    required this.originalLanguage,
+    required this.popularity,
+    required this.hasVideo,
+    required this.voteCount,
+    required this.isAdult,
+  });
+
+  MovieEntity.fromMovieModel(MovieModel movieModel)
```

Comment:
Maybe you can create a factory method here like
`factory MovieEntity.fromModel({required MovieModel model}) => MovieEntity(...)`